creator/ingest_requests/models.py

```diff
@@ -1,16 +1,160 @@
 from django.db import models
 
+from creator.files.models import Version
+from creator.jobs.models import Job
+
+DELIMITER = "-"
+NAME_PREFIX = "INGEST_REQUEST"
+
+User = get_user_model()
+
 
 class IngestRequest(models.Model):
     """
-    Plug the model in here
+    Request to ingest file(s) into a target data service
     """
 
     class Meta:
         permissions = [
             ("list_all_ingestrequest", "Show all ingest_requests"),
         ]
 
+    name = models.CharField(
+        max_length=500,
+        unique=True,
+        blank=True,
+        null=True,
+        help_text=(
+            "The name of the ingest request. An field that is auto populated "
+            "by concatenating the IDs of the IngestRequest's versions"
+        ),
+    )
+
+    versions = models.ManyToManyField(
+        Version,
+        related_name="ingest_requests",
+        help_text="List of files to ingest in the ingest request",
+    )
+
+    job = models.OneToOneField(
+        Job,
+        null=True,
+        blank=True,
+        related_name="ingest_request",
+        on_delete=models.CASCADE,
+        help_text="The associated ingest job for this ingest request",
+    )
+
     created_at = models.DateTimeField(
-        null=False, help_text="Time when the ingest_request was created"
+        auto_now_add=True,
+        null=True,
+        help_text="Time when the ingest request was created",
+    )
+
+    creator = models.ForeignKey(
+        User,
+        null=True,
+        blank=True,
+        related_name="ingest_requests",
+        on_delete=models.SET_NULL,
+        help_text="The user who submitted this ingest request",
     )
+
+    @staticmethod
+    def name_from_version_ids(version_ids):
+        """
+        Create IngestRequest.name by joining the version ids together
+        """
+        return DELIMITER.join([NAME_PREFIX] + version_ids)
+
+    @staticmethod
+    def version_ids_from_name(request_name):
+        """
+        Parse IngestRequest.name into the file version ids which were used
+        to create the name
+        """
+        return request_name.lstrip(NAME_PREFIX).split(DELIMITER)
+
+    @classmethod
+    def find_by_version_ids(cls, version_ids):
+        """
+        Lookup an IngestRequest given a list of Version.kf_id strings
+        """
+        return cls.objects.filter(
+            name=cls.name_from_version_ids(version_ids)
+        ).first()
+
+    def start_job(self, task, *args, **kwargs):
+        """
+        Enqueue an ingest task to start an ingest job
+        """
+        pass
+
+    def cancel_job(self):
+        """
+        Cancel or stop the associated job if it is running
+
+        This method is called when an IngestRequest is deleted
+        """
+        pass
+
+    def clean(self):
+        """
+        Validate model instance and raise a
+        django.core.exceptions.ValidationError if validation fails
+        """
+        if not self._versions_exist():
```

Comment:
It seems like this could more easily be replaced by `if not self.versions.exists():`